cmd/singularity/actions_test.go

```diff
@@ -319,6 +323,138 @@ func testRunFromURI(t *testing.T) {
+	cmd.Stdout = &out
+	err = cmd.Run()
+	if err != nil {
+		t.Fatal(err)
+	}
+	cmd = exec.Command("mkfs.ext3", "-q", "-F", "ext3_fs.img")
+	cmd.Stdout = &out
+	err = cmd.Run()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove("ext3_fs.img")
+
+	// create a file dir
+	t.Run("overlay_create", test.WithPrivilege(func(t *testing.T) {
+		_, stderr, exitCode, err := imageExec(t, "exec", opts{overlay: []string{dir}}, imagePath, []string{"touch", "/dir_overlay"})
+		if exitCode != 0 {
+			t.Log(stderr, err)
+			t.Fatalf("unexpected failure running '%v'", strings.Join([]string{"test", "-f", "/dir_overlay"}, " "))
+		}
+	}))
+	// look for the file dir
+	t.Run("overlay_find", test.WithPrivilege(func(t *testing.T) {
+		_, stderr, exitCode, err := imageExec(t, "exec", opts{overlay: []string{dir}}, imagePath, []string{"test", "-f", "/dir_overlay"})
+		if exitCode != 0 {
+			t.Log(stderr, err)
+			t.Fatalf("unexpected failure running '%v'", strings.Join([]string{"test", "-f", "/dir_overlay"}, " "))
+		}
+	}))
+	// create a file ext3
+	t.Run("overlay_ext3_create", test.WithPrivilege(func(t *testing.T) {
+		_, stderr, exitCode, err := imageExec(t, "exec", opts{overlay: []string{"ext3_fs.img"}}, imagePath, []string{"touch", "/ext3_overlay"})
+		if exitCode != 0 {
+			t.Log(stderr, err)
+			t.Fatalf("unexpected failure running '%v'", strings.Join([]string{"test", "-f", "/ext3_overlay"}, " "))
+		}
+	}))
+	// look for the file ext3
+	t.Run("overlay_ext3_find", test.WithPrivilege(func(t *testing.T) {
+		_, stderr, exitCode, err := imageExec(t, "exec", opts{overlay: []string{"ext3_fs.img"}}, imagePath, []string{"test", "-f", "/ext3_overlay"})
+		if exitCode != 0 {
+			t.Log(stderr, err)
+			t.Fatalf("unexpected failure running '%v'", strings.Join([]string{"test", "-f", "/ext3_overlay"}, " "))
+		}
+	}))
+	// look for the file squashFs
+	t.Run("overlay_squashFS_find", test.WithPrivilege(func(t *testing.T) {
+		_, stderr, exitCode, err := imageExec(t, "exec", opts{overlay: []string{squashfsImage}}, imagePath, []string{"test", "-f", fmt.Sprintf("/%s", tmpfile.Name())})
+		if exitCode != 0 {
+			t.Log(stderr, err)
+			t.Fatalf("unexpected failure running '%v'", strings.Join([]string{"test", "-f", fmt.Sprintf("/%s", tmpfile.Name())}, " "))
+		}
+	}))
+	// create a file multiple overlays
+	t.Run("overlay_multiple_create", test.WithPrivilege(func(t *testing.T) {
+		_, stderr, exitCode, err := imageExec(t, "exec", opts{overlay: []string{"ext3_fs.img", squashfsImage}}, imagePath, []string{"touch", "/multiple_overlay_fs"})
+		if exitCode != 0 {
+			t.Log(stderr, err)
+			t.Fatalf("unexpected failure running '%v'", strings.Join([]string{"touch", "/multiple_overlay_fs"}, " "))
+		}
+	}))
+	// look for the file with multiple overlays
+	t.Run("overlay_multiple_find", test.WithPrivilege(func(t *testing.T) {
+		_, stderr, exitCode, err := imageExec(t, "exec", opts{overlay: []string{"ext3_fs.img", squashfsImage}}, imagePath, []string{"test", "-f", "/multiple_overlay_fs"})
```

Comment:
Maybe check presence of squash file too with `[]string{"test", "-f", fmt.Sprintf("/%s", tmpfile.Name())}`